File: lian/week6/3-1904.py
import sys
input = sys.stdin.readline
n = int(input())
f = [0] * 1000001
f[0] = 1
f[1] = 2
for i in range(2,n):
    f[i] = (f[i-1]+f[i-2])%15746
print(f[n-1])

File: leo/week6/3-1904.js
/**
 *  제목 :
 *  입력 :
 *  제한사항 :
 *  시간계산 필요 :
 */

// Run by Node.js
const readline = require("readline");

(async () => {
  let rl = readline.createInterface({ input: process.stdin });

  let n;
  for await (const line of rl) {
    n = parseInt(line);
  }
  rl.close();

  let dp = [0, 1, 2];
  for (let i = 3; i <= n; ++i) {
    dp[i] = (dp[i - 2] + dp[i - 1]) % 15746;
  }

  console.log(dp[n]);

  process.exit();
})();
